### PR TITLE
update terraform hetzner dns

### PR DESCRIPTION
**Pull request by rwaffen**
Updateing environment and adding hetzner dns with example42.cloud domain

### CODE REVIEW

**Comment by SimonHoenscheid** [comment]
LGTM

File: bin/bootstrap/cloud_init.sh [hunk 1]
```diff
@@ -1,37 +1,43 @@
 #!/bin/bash
-echo -e "10.0.1.1 puppet\n10.0.1.2 gitlab\n" >> /etc/hosts
+
 puppet_version=$1
 puppet_role=$2
 control_repo=$3
 
-yum install -y git
-git clone $control_repo /tmp/psick
+echo "### Installing git"
```
Comment (alvagante) [inline bin/bootstrap/cloud_init.sh:7]: A nice touch would be to run which git before to check is we need to install it.

File: bin/puppet_install.sh [hunk 2]
```diff
@@ -40,85 +44,80 @@ fi
...
-    8) codename=jessie  ;;
+    8) codename=jessie   ;;
     9) codename=stretch  ;;
-    10) codename=buster ;;
-    14.04) codename=trusty  ;;
+    10) codename=buster  ;;
+    14.04) codename=trusty ;;
     16.04) codename=xenial ;;
     18.04) codename=bionic ;;
-    18.10) codename=bionic ;;
```
Comment (alvagante) [inline bin/puppet_install.sh:105]: why are these removed? Should work the same with the packages packed for lts releases

File: terraform/variables.tf [hunk 3]
```diff
@@ -35,23 +36,22 @@ variable "machines" {
...
-    "student9"  = { ip = "10.0.1.19", role = "student", server_type = "cx11", access_level = "all_keys" }
-    "student10" = { ip = "10.0.1.20", role = "student", server_type = "cx11", access_level = "all_keys" }
-    "student11" = { ip = "10.0.1.21", role = "student", server_type = "cx11", access_level = "all_keys" }
-    "student12" = { ip = "10.0.1.22", role = "student", server_type = "cx11", access_level = "all_keys" }
-    "student13" = { ip = "10.0.1.23", role = "student", server_type = "cx11", access_level = "all_keys" }
-    "student14" = { ip = "10.0.1.24", role = "student", server_type = "cx11", access_level = "all_keys" }
-    "student15" = { ip = "10.0.1.25", role = "student", server_type = "cx11", access_level = "all_keys" }
+    "puppet"    = { ip = "10.0.1.1",  role = "puppet",  server_type = "cx41", access_level = "admin_keys", image = "centos-7" }
+    "gitlab"    = { ip = "10.0.1.2",  role = "gitlab",  server_type = "cx21", access_level = "admin_keys", image = "centos-7" }
+    "student1"  = { ip = "10.0.1.11", role = "student", server_type = "cx11", access_level = "all_keys",   image = "centos-7" }
```
Comment (alvagante) [inline terraform/variables.tf:41]: it could be nice to be able to customize with a single number the amount of student vms to create (not priority)

**Review by alvagante — APPROVED** [review]
good job

File: bin/puppet_install.sh (hunk 2 above)
Comment (rwaffen) [reply bin/puppet_install.sh:105]: i thought because its EOL….

File: terraform/variables.tf (hunk 3 above)
Comment (rwaffen) [reply terraform/variables.tf:41]: you can overwrite the hash and so customize it. its only default data.

**Review by SimonHoenscheid — APPROVED** [review]

File: terraform/variables.tf [hunk 4]
```diff
@@ -23,7 +24,7 @@ variable "control_repo" {
 
 variable "puppet_version" {
   type = number
-  description = "The Puppet version to use. Specify 5,6 or 7"
+  description = "The Puppet version to use. Specify 5, 6 or 7"
   default = 6
```
Comment (tuxmea) [inline terraform/variables.tf:28]: We should default to puppet 7.

**Review by tuxmea — CHANGES_REQUESTED** [review]
